### PR TITLE
[DRAFT] refactor accuracy check and move graph_comparator (#72470)

### PR DESCRIPTION
**Pull request by SmirnovGrigory**
### Done:
 - *refactored accuracy check function to use test util infrastructure*
 - *moved graph_comparator to common_test_utils target*
 - *add tests for graph_comparator*

 ### TODO:
 - *add tests of new accuracy check*
 - *make accuracy check a part of graph_comparator*
 - *decide in which tests we want to enable accuracy check*

### Tickets:
 - *72470*

### CODE REVIEW

**Comment by SmirnovGrigory** [comment]
image

**Comment by SmirnovGrigory** [comment]
image

**Comment by SmirnovGrigory** [comment]
image

**Comment by SmirnovGrigory** [comment]
image

File: src/tests/ie_test_utils/common_test_utils/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ function(add_common_utils ADD_TARGET_NAME)
             ROOT ${CMAKE_CURRENT_SOURCE_DIR}
             EXCLUDED_SOURCE_PATHS
                 ${CMAKE_CURRENT_SOURCE_DIR}/gtest
+                ${CMAKE_CURRENT_SOURCE_DIR}/tests/
```
Comment (GlebKazantaev) [inline src/tests/ie_test_utils/common_test_utils/CMakeLists.txt:13]:
```suggestion
                # FIXME: number_of_ticket
                ${CMAKE_CURRENT_SOURCE_DIR}/tests/
```

File: src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp [hunk 2]
```diff
@@ -5,235 +5,31 @@
...
+        for (const auto& param : ref_function->get_parameters()) {
+            const auto &tensor = ov::test::utils::create_and_fill_tensor(param->get_element_type(),
+                                                                        param->get_shape());
+            input_data[param] = tensor;
         }
 
-        auto ref_outputs = ngraph::helpers::interpreterFunction(ref_function, input_data, types);
-        auto outputs = ngraph::helpers::interpreterFunction(cur_function, input_data, types);
+        auto ref_outputs = ngraph::helpers::interpretFunction(ref_function, input_data);
+        auto outputs = ngraph::helpers::interpretFunction(ref_function, input_data);
```
Comment (GlebKazantaev) [inline src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:25]: Looks like we compare inference for two the same functions here.

Comment (GlebKazantaev) [reply src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp:25]: Also I think input data must be copied for cur_function

Comment (SmirnovGrigory) [reply src/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp]: my mistake, corrected

File: src/tests/ie_test_utils/common_test_utils/CMakeLists.txt (hunk 1 above)
Comment (SmirnovGrigory) [reply src/tests/ie_test_utils/common_test_utils/CMakeLists.txt:14]: done

**Review by GlebKazantaev — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**